### ISSUE TITLE
Play Console: Segment Non-Compliant SDK Version

### ISSUE THREAD

**Issue by huhx0015** [fullstorydev/fullstory-segment-middleware-android#24]
Hello,

Recently, we have been receiving warnings by the Google Play Console to update the Segment.io dependency to a compliant version. We've narrowed down the cause of this issue and it appears that [FullStory-Segment-Middleware-Android](https://github.com/fullstorydev/fullstory-segment-middleware-android) is using a non-compliant version of [Segment (4.7.1)](https://github.com/fullstorydev/fullstory-segment-middleware-android/blob/f7b237fedf947baddc4fa3f1ed8223b329cc34e7/fullstory-segment-middleware/build.gradle#L42).

<img width="1323" alt="Screen Shot 2022-07-19 at 11 46 16 AM" src="https://user-images.githubusercontent.com/1645482/179826760-9fd0ff21-d2d7-4b76-b8b9-a2c8c1bafe8e.png">

It appears that any app that uses this non-compliant version of the Segment.io dependency (which Fullstory-Segment-Middleware-Android includes) will be blocked after August 9, 2022.

Segment has already resolved this issue, in Segment 4.10.1 and above: https://github.com/segmentio/analytics-android/issues/802

Are there any plans to update this dependency to resolve this issue?

**Comment by prashanthsadasivan-fullstory** [comment]
Hey @huhx0015, we're not well equipped to handle support via GitHub Issues - I'd recommend sending questions like this through `mobile-support@fullstory.com` for a more expedited response. 

In this case, I took a quick look, and it looks like the library _is actually_ specifying [4.7.1](https://github.com/fullstorydev/fullstory-segment-middleware-android/blob/master/fullstory-segment-middleware/build.gradle#L42), and not 4.8.2 as in your screenshot. I took a look at our demo app in the repo and ran a `./gradlew demo:dependencies`, and it showed a dependency on the version [4.9.0](https://github.com/fullstorydev/fullstory-segment-middleware-android/blob/master/demo/build.gradle#L46) that our demo application specified. My guess is that your application is specifying the 4.8.2 somewhere in your dependency list. I'd recommend using gradle to figure out where that dependency is coming from. 

If that doesn't help you track down where that version is coming from, email mobile-support@fullstory.com and we can open a ticket to triage this issue.

**Comment by prashanthsadasivan-fullstory** [comment]
Yup, I saw those issues. I've got two things I'm confused on

1) Your application's version should be overriding any other version if it's the latest and is compatible with anything else depending on it.
2) The library's version is 4.7.1, not 4.8.2 - where is the 4.8.2 coming from? did Envoy's App version code 83 (from the text of the email) use `4.8.2`?

**Comment by huhx0015** [comment]
Thanks @prashanthsadasivan-fullstory! Investigating into this more, I can confirm that the Segment dependency used in `fullstory-segment-middleware-android` is definitely being overridden by the latest version of the Segment dependency that we are declaring. 

The Google Play warnings are actually a false alarm, as we found out that this is coming from an older version of our app published on forgotten, unused track. This version definitely used Segment `4.8.2`, but the warnings would be thrown by Google Play every time we've published a new update, making it appear that the latest app update was at issue. 🤦🏻‍♂️ 

Either way, this no longer seems to be an issue. Thanks for your assistance @prashanthsadasivan-fullstory and also apologies for making this a bigger issue than it really is. 😓
